### PR TITLE
Update and refactor the current example

### PR DESCRIPTION
**Pull request by eperinan**
These following points are treated in this PR.

- [x] Bumps Freestyle version to `0.7.0`.
- [x] Bumps Freestyle-rpc version to `0.11.0`.
- [x] Use tagless against frees.
- [x] Removes Future client and replace with IO client as a pure effect.
- [x] Uses log4s with logbackClassic to log messages.

### CODE REVIEW

**Review by raulraja — APPROVED** [review]
Nice cleanup 👍

File: build.sbt [hunk 1]
```diff
@@ -1,8 +1,10 @@
 lazy val V = new {
-  lazy val frees    = "0.4.1"
-  lazy val freesRPC = "0.1.2"
-  lazy val circe    = "0.9.0-M1"
-  lazy val monix    = "3.0.0-M1"
+  lazy val circe          = "0.9.1"
+  lazy val frees          = "0.7.0"
+  lazy val freesRPC       = "0.11.0"
```
Comment (juanpedromoreno) [inline build.sbt]: It could be 0.11.1, not a big deal though.

File: routeguide/src/main/scala/handlers/RouteGuideClientHandler.scala [hunk 2]
```diff
@@ -2,23 +2,31 @@ package routeguide
...
 import cats.implicits._
 import io.grpc.{Status, StatusRuntimeException}
-import journal.Logger
 import monix.eval.Task
 import monix.reactive.Observable
+import org.log4s._
 import routeguide.protocols._
+import freestyle.tagless._
+import freestyle.tagless.async.implicits._
+import freestyle.tagless.config.implicits._
```
Comment (juanpedromoreno) [inline routeguide/src/main/scala/handlers/RouteGuideClientHandler.scala]: I don't think this import is needed here.

File: build.sbt [hunk 3]
```diff
@@ -12,14 +14,21 @@ lazy val `demo-routeguide` = project
   .settings(Seq(
     organization := "frees-io",
     organizationName := "47 Degrees",
-    scalaVersion := "2.12.3",
+    scalaVersion := "2.12.4",
     resolvers ++= Seq(
       Resolver.sonatypeRepo("releases"),
       Resolver.bintrayRepo("beyondthelines", "maven")
     ),
+    scalacOptions := Seq("-Ywarn-unused-import"),
```
Comment (juanpedromoreno) [inline build.sbt]: This is overriding all the existing `scalacOptions`, let's change it for on `+=`.

**Review by juanpedromoreno — APPROVED** [review]
LGTM, thanks @eperinan !

File: build.sbt (hunk 3 above)
Comment (eperinan) [reply build.sbt]: Addressed. Thanks for you help

**Pull request merged**